crates/ide_assists/src/handlers/replace_derive_with_manual_impl.rs

```diff
@@ -52,18 +53,34 @@ pub(crate) fn replace_derive_with_manual_impl(
         return None;
     }
 
-    let trait_token = args.syntax().token_at_offset(ctx.offset()).find(|t| t.kind() == IDENT)?;
-    let trait_name = trait_token.text();
-
+    let current_derives = args.clone().as_comma_sep_paths()?;
+    // found the related path of current focused token
+    // I don't know how to keep the original range for tokens while transforming and parsing as new ast
```

Comment:
You could iterate through the token tree of the attribute and count the number of commas until you find the node of the cursor in it. That number of commas is then the index into `current_derives`. This should always work unless the user screwed up the derive syntax at which point its fine for this to not work properly.

Reply:
I think only counting commas count possibly result in the assist applicable in extra white space? I decided to reuse part of the logic in https://github.com/rust-analyzer/rust-analyzer/pull/10847/files#diff-0e0e5f4792d3e8922d28bb09e751cecc255acc261d4a62e126fd9bcdf00980f8R56-R69

Reply:
sorry I think I missed up the rebasing.. fixing it now..

Reply:
Oh nice find, I forgot about us doing something similar there already 👍